src/TBuffer.cpp

```diff
@@ -30,6 +29,552 @@
 
 #include <assert.h>
 
+// Define this to get qDebug() messages about the decoding of UTF-8 data when it
+// is not the single bytes of pure ASCII text:
+#define DEBUG_UTF8_PROCESSING
+
+// These tables have been regenerated from examination of the Qt source code
+// particularly from:
+// https://gitorious.org/qt/qt?p=qt:qt.git;a=blob;f=src/corelib/codecs/qsimplecodec.cpp;h=cb52ce35f369f7fbe5b04ff2c2cf1600bd794f4e;hb=HEAD
+// It represents the Unicode codepoints that are to be used to represent
+// extended ASCII characters with the MS Bit set.  The name is one that will
+// be recognized as a valid encoding name to supply to:
+// QTextCodec::codecForName(...)
+// "ISO 885901" is not included here as it is inherent in Qt and has a straight
+// one for one mapping of characters in the range 128 to 255 to the
+// corresponding Unicode codepoint.
+// Note that the codepoint 0xFFFD is the Unicode replacement character and
+// is reserved to show, amongst other things, where a UTF-8 sequence of bytes
+// is not a known character or, in these tables, the fact that no character was
+// defined at that Extended ASCII character code.
+
+// clang-format off
+// Do not let code reformatting tool mess this around!
+// PLACEMARKER: Extended ASCII decoder Unicode codepoint lookup tables
+const QMap<QString, QVector<QChar>> TBuffer::csmEncodingTable = {
```

Comment:
That is initialising a __single__, __global__, __static__ instance of the `TBuffer::csmEncodingTable` I thought that __has__ to take place in the __implementation__ file so that there is only one copy in the whole linked application?